x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/deprecation/DeprecationIssue.java

```diff
@@ -175,4 +190,233 @@ public int hashCode() {
     public String toString() {
         return Strings.toString(this);
     }
+
+    public static Map<String, Object> createMetaMapForRemovableSettings(List<String> removableSettings) {
+        return Meta.fromRemovableSettings(removableSettings).toMetaMap();
+    }
+
+    /**
+     * This method returns a DeprecationIssue that has in its meta object the intersection of all auto-removable settings that appear on
+     * all of the DeprecationIssues that are passed in. This method assumes that all DeprecationIssues passed in are equal, except for the
+     * auto-removable settings in the meta object.
+     * @param similarIssues DeprecationIssues that are assumed to be identical except possibly removal actions.
+     * @return A DeprecationIssue containing only the removal actions that are in all similarIssues
+     */
+    public static DeprecationIssue getIntersectionOfRemovableSettings(List<DeprecationIssue> similarIssues) {
+        if (similarIssues == null || similarIssues.isEmpty()) {
+            return null;
+        }
+        if (similarIssues.size() == 1) {
+            return similarIssues.get(0);
+        }
+        DeprecationIssue representativeIssue = similarIssues.stream().findAny().get();
```

Comment:
Seems like this could be `similarIssues.get(0)` since it's not filtering or anything?

Reply:
Yeah that's what I originally had it as, but I changed it to make it clear that it could be any (the first one's not special). I can put it back. I flip-flopped a couple of times on this and am not tied to either way.